src/main/java/com/google/devtools/build/remote/client/AbstractRemoteActionCache.java

```diff
@@ -56,7 +57,12 @@ public AbstractRemoteActionCache(DigestUtil digestUtil) {
    *     failed.
    */
   public Tree getTree(Digest rootDigest) throws IOException {
-    Directory rootDir = Directory.parseFrom(downloadBlob(rootDigest));
+    Directory rootDir;
+    try {
+      rootDir = Directory.parseFrom(downloadBlob(rootDigest));
+    } catch (IOException e) {
```

Comment:
This kind of exception handling seems strange to me, though this might be from my lack on understanding of Java. Ola, could you correct me if I'm wrong?

Does this add anything that one couldn't get from the stack trace? I understand when you're adding a filename to the message, I'm not sure why this is needed otherwise.

Reply:
IIUC, in this handler you're conflating exceptions with failed download (as in "NOT_FOUND") and exception with parsing the result as a particular proto (InvalidProtocolBufferException which extends IOException). In both cases, the message should be informative, and the wrapper does add information, I think. Maybe consider splitting these cases?

Reply:
@agoulti 
My intention here (and the other places where I added this sort of handling) was to provide some more information for when the IOException occurred. Without it, it will just show something like "Missing Digest XXXXX/XX" along with a stack trace. I thought maybe reading a stack trace wasn't ergonomic enough for a debugging tool so I wanted to provide a more clear error message on top of it to provide some context. In particular, you might need to infer what type of blob was being downloaded. I'm also fine with just providing the stack trace though, as in most cases it's pretty obvious.

Also I just noticed that in addChildDirectories (a recursive call to grab all the Directories in a Tree), I throw this, causing the exception to get wrapped at every parent subdirectory, so I will at least remove that.

@ola-rozenfeld 
I actually didn't intend to conflate InvalidProtocolBufferException, as I thought it wasn't an IOException. I will separate it.

Reply:
Hi, I've made changes to address these comments.

I didn't end up splitting the cases because I think it's clear enough whether the exception was caused by parsing or download or whatever else by the error message of the inner exception. However, I changed some of the other exceptions to say 'obtain' rather than 'download' to make it more clear that a parsing error could have occurred.